azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/SdkTypeHydrator.java

```diff
@@ -0,0 +1,14 @@
+package com.microsoft.azure.functions.sdktype;
+
+/**
+ * A hydrator that builds the final client instance
+ * from a given SdkTypeMetaData object.
+ *
+ * @param <M> the type of MetaData used
+ */
+public interface SdkTypeHydrator<M extends SdkTypeMetaData> {
+    /**
+     * Build the final object using data from the metaData.
+     */
```

Comment:
Why are we returning generic Object from the createInstance? What type you are actually expecting from createInstance.